### PR TITLE
Fix some bugs

### PR DESCRIPTION
**Pull request by jordane95**
* Now dp should be the third dimension of parallel ranks
* One missing argument

### CODE REVIEW

File: src/nanotron/serialize/weights.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ def save_weights(model: nn.Module, parallel_context: ParallelContext, root_folde
 
             else:
                 exp_tp_pp_rank_and_size = None
+                is_expert_sharded = False
```
Comment (xrsrke) [inline src/nanotron/serialize/weights.py:88]: why do we put is_expert_sharded = False? would be cool if we put some notes here

**Review by xrsrke — CHANGES_REQUESTED** [review]

**Review by xrsrke — CHANGES_REQUESTED** [review]
> Now dp should be the third dimension of parallel ranks
Could you please elaborate on this?

> One missing argument
Where is it 🤔

**Comment by jordane95** [comment]
> > Now dp should be the third dimension of parallel ranks
> 
> Could you please elaborate on this?
> 
> > One missing argument
> 
> Where is it 🤔

1. It is quite straightforward by looking at the definition of rank matrix
https://github.com/huggingface/nanotron/blob/ff3c7746577948743da08c4868aca46cbc0c110b/src/nanotron/parallel/context.py#L69-L76

2. Missing `is_expert_sharded` variable would cause referenced before defined error

https://github.com/huggingface/nanotron/blob/ff3c7746577948743da08c4868aca46cbc0c110b/src/nanotron/serialize/weights.py#L72-L95

**Review by NouamaneTazi — APPROVED** [review]
LGTM! it's part of my series of expert parallelism fixes. Some fixes made it in a previous PR. Thanks for noticing!

**Pull request merged**